uitest/Extensions/WindowsDriverExtensions.cs

```diff
@@ -0,0 +1,139 @@
+    /// <param name="pollingInterval">Action interval</param>
+    /// <returns>Instance of <see cref="DefaultWait{T}"/></returns>
+    public static DefaultWait<WindowsDriver<WindowsElement>> Wait(
+        this WindowsDriver<WindowsElement> driver,
+        TimeSpan timeout,
+        TimeSpan pollingInterval)
+    {
+        return new (driver)
+        {
+            Timeout = timeout,
+            PollingInterval = pollingInterval,
+        };
+    }
+
+    /// <summary>
+    /// Create an instance of <see cref="DefaultWait{T}"/>
+    /// </summary>
+    /// <param name="driver">Driver instance</param>
+    /// <param name="timeoutInMS">Wait timeout in milliseconds</param>
+    /// <param name="pollingIntervalInMS">Action interval in milliseconds</param>
+    /// <returns>Instance of <see cref="DefaultWait{T}"/></returns>
+    public static DefaultWait<WindowsDriver<WindowsElement>> Wait(
+        this WindowsDriver<WindowsElement> driver,
+        int timeoutInMS = DefaultTimeoutInMS,
+        int pollingIntervalInMS = DefaultPollingIntervalInMS)
+    {
+        return driver.Wait(TimeSpan.FromMilliseconds(timeoutInMS), TimeSpan.FromMilliseconds(pollingIntervalInMS));
+    }
+
+    /// <summary>
+    /// Create an instance of <see cref="DefaultWait{T}"/> which ignores all <see cref="WebDriverException"/> exceptions
+    /// </summary>
+    /// <param name="driver">Driver instance</param>
+    /// <param name="timeoutInMS">Wait timeout in milliseconds</param>
+    /// <param name="pollingIntervalInMS">Action interval in milliseconds</param>
+    /// <returns>Instance of <see cref="DefaultWait{T}"/></returns>
+    public static DefaultWait<WindowsDriver<WindowsElement>> WaitAndIgnoreExceptions(
+        this WindowsDriver<WindowsElement> driver,
+        int timeoutInMS = DefaultTimeoutInMS,
+        int pollingIntervalInMS = DefaultPollingIntervalInMS)
+    {
+        var wait = driver.Wait(timeoutInMS, pollingIntervalInMS);
+        wait.IgnoreExceptionTypes(typeof(WebDriverException));
+        return wait;
+    }
+
+    /// <summary>
+    /// Wait until an element is invisible or timeout
+    /// </summary>
+    /// <param name="driver">Driver instance</param>
+    /// <param name="by">Element locator</param>
+    /// <param name="timeoutInMS">Wait timeout in milliseconds</param>
+    /// <param name="pollingIntervalInMS">Action interval in milliseconds</param>
+    public static void WaitUntilInvisible(
+        this WindowsDriver<WindowsElement> driver,
+        By by,
+        int timeoutInMS = DefaultTimeoutInMS,
+        int pollingIntervalInMS = DefaultPollingIntervalInMS)
+    {
+        driver
+            .Wait(timeoutInMS, pollingIntervalInMS)
+            .Until(_ =>
+            {
+                try
+                {
+                    return !driver.FindElement(by).Displayed;
+                }
+                catch (WebDriverException)
+                {
+                    return true;
+                }
```

Comment:
Why do you need to catch an exception here?
Could you do something like `return driver?.FindElement(by)?.Displayed != true`?

Reply:
No, `FindElement` will throw an exception if the element was not found.